Sources/MapboxMaps/Foundation/Camera/CameraTransition.swift

```diff
@@ -0,0 +1,113 @@
+              let renderedPadding = renderedCameraOptions.padding,
+              let renderedPitch = renderedCameraOptions.pitch,
+              let renderedBearing = renderedCameraOptions.bearing else {
+            fatalError("Values in rendered CameraOptions cannot be nil")
+        }
+
+        center  = .init(fromValue: renderedCenter)
+        zoom    = .init(fromValue: renderedZoom)
+        padding = .init(fromValue: renderedPadding)
+        pitch   = .init(fromValue: renderedPitch)
+        bearing = .init(fromValue: renderedBearing)
+        anchor  = .init(fromValue: initialAnchor)
+    }
+
+    internal var toCameraOptions: CameraOptions {
+        return CameraOptions(center: center.toValue,
+                             padding: padding.toValue,
+                             anchor: anchor.toValue,
+                             zoom: zoom.toValue,
+                             bearing: shouldOptimizeBearingPath ? Self.optimizeBearing(startBearing: bearing.fromValue,
+                                                                                       endBearing: bearing.toValue)
+                                                                :  bearing.toValue,
+                             pitch: pitch.toValue)
+    }
+
+    internal var fromCameraOptions: CameraOptions {
+        return CameraOptions(center: center.fromValue,
+                             padding: padding.fromValue,
+                             anchor: anchor.fromValue,
+                             zoom: zoom.fromValue,
+                             bearing: bearing.fromValue,
+                             pitch: pitch.fromValue)
+
+    }
+
+    /// This function optimizes the bearing for set camera so that it is taking the shortest path.
+    /// - Parameters:
+    ///   - startBearing: The current or start bearing of the map viewport.
+    ///   - endBearing: The bearing of where the map viewport should end at.
+    /// - Returns: A `CLLocationDirection` that represents the correct final bearing accounting for positive and negatives.
+    internal static func optimizeBearing(startBearing: CLLocationDirection?, endBearing: CLLocationDirection?) -> CLLocationDirection? {
+        // This modulus is required to account for larger values
+        guard
+            let startBearing = startBearing?.truncatingRemainder(dividingBy: 360.0),
+            let endBearing = endBearing?.truncatingRemainder(dividingBy: 360.0)
+        else {
+            return nil
+        }
+
+        // 180 degrees is the max the map should rotate, therefore if the difference between the end and start point is
+        // more than 180 we need to go the opposite direction
+        if endBearing - startBearing >= 180 {
+            return endBearing - 360
+        }
+
+        // This is the inverse of the above, accounting for negative bearings
+        if endBearing - startBearing <= -180 {
+            return endBearing + 360
+        }
+
+        return endBearing
+    }
```

Comment:
I don't think this produces the correct results. Consider this example:

    bearing.fromValue = -560
    bearing.toValue = 0

as written, we'll end up with

    fromCameraOptions.bearing = -560
    toCameraOptions.bearing = 0

which can't be right

Reply:
let's update with a test that fails for this case before fixing

Reply:
Hmm this was just moved from what was already present in `CameraManager`.. so it's an existing bug.

Reply:
Wait not sure if there is a bug..  

If we pass in 
```
bearing.fromValue = -560
bearing.toValue = 0
```

we would end up with 
```swift
fromCameraOptions.bearing = -560 //since `fromValues` don't get mutated
toCameraOptions.bearing = -360
```
Which should result in the right behavior.. But there's an argument to be made that we work only with "normalized" values.

Reply:
Here's what I see if I run this code:

```swift
        mapView.on(.styleLoaded) { [weak self] _ in
            guard let self = self else { return }
            // Center the map over New York City.
            self.mapView.camera.setCamera(to: CameraOptions(center: self.newYork, zoom: 14, bearing: -560))
        }

        // Allows the delegate to receive information about map events.
        mapView.on(.mapLoaded) { [weak self] _ in
            guard let self = self else { return }

            self.animator = mapView.camera.makeAnimator(duration: 4, curve: .easeInOut) { (transition) in
                transition.bearing.fromValue = -560
                transition.bearing.toValue = 0
            }

            self.animator.addCompletion { (_) in
                print("All animations complete!")
            }
            
            self.finish()
        }
```


https://user-images.githubusercontent.com/6844889/115792227-e4dcd900-a397-11eb-8a83-e81847eaa301.MP4

Reply:
From discussion: try removing truncation on the start value

Reply:
This is done.